products/views.py

```diff
@@ -1,3 +1,108 @@
-from django.shortcuts import render
+from django.http     import JsonResponse
+from django.http.response import Http404
+from django.views    import View
 
-# Create your views here.
+from products.models import *
+
+class CategoryView(View) :
+  def get(self, request) :
+
+    category_list = [{
+      'id'             : gender_category.id,
+      'name'           : gender_category.name,
+      'main_category'  : [{
+        'id'   : main_category.id,
+        'name' : main_category.name,
+        'sub_category' : [{
+          'id'   : sub_category.id,
+          'name' : sub_category.name
+        } for sub_category in main_category.subcategory_set.all()]
+      } for main_category in gender_category.maincategory_set.all()]
+    } for gender_category in GenderCategory.objects.all().order_by('id')]
+
+    return JsonResponse({'category_list':category_list}, status=200)
+
+class ProductView(View) :
+  def get(self, request) :
+
+    order_request = request.GET.get('sort')
+    
+    SORT_PREFIX = {
+      'recent'    : '-created_at',
+      'old'       : 'created_at',
+      'ascPrice'  : 'price',
+      'descPrice' : '-price'
+    }
+
+    FILTER_PREFIX = {
+      'gender'        : 'sub_category__main_category__gender_category__in',
+      'main_category' : 'sub_category__main_category__in',
+      'sub_category'  : 'sub_category__in',
+      'color'         : 'color__in',
+      'size'          : 'productsize__size__in',
+      'collection'    : 'collection__in',
+      'conscious'     : 'is_conscious__in'
+    }
+
+    filter_set = {
+      FILTER_PREFIX.get(key) : value for (key, value) in dict(request.GET).items() if FILTER_PREFIX.get(key)
+    }
+
+    try :
+      products = Product.objects.filter(**filter_set).order_by(SORT_PREFIX[order_request])
+
+      products_list = [{
+        'id'            : product.id,
+        'name'          : product.name,
+        'price'         : format(int(product.price), ','),
+        'is_new'        : product.is_new,
+        'is_conscious'  : product.is_conscious,
+        'collection_id' : product.collection_id,
+        'color'         : [i.color for i in products.filter(name=product.name)],
+        'image'         : [i.url for i in product.image_set.all()][0:2]
```

Comment:
변수명을 명확하게 선언해주세요. `복수인지 단수인지`, `i` 는 피해주세요.
```suggestion
        'images'         : [image.url for image in product.image_set.all()][0:2]
```